vendor/bat-native-ledger/src/bat/ledger/ledger_database.cc

```diff
@@ -4,12 +4,349 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #include "bat/ledger/ledger_database.h"
-#include "bat/ledger/internal/ledger_database_impl.h"
+
+#include <utility>
+#include <vector>
+
+#include "base/bind.h"
+#include "base/files/file_util.h"
+#include "bat/ledger/internal/logging/logging.h"
+#include "sql/statement.h"
+#include "sql/transaction.h"
 
 namespace ledger {
 
-LedgerDatabase* LedgerDatabase::CreateInstance(const base::FilePath& path) {
-  return new LedgerDatabaseImpl(path);
+namespace {
+
+void HandleBinding(sql::Statement* statement,
+                   const mojom::DBCommandBinding& binding) {
+  if (!statement) {
+    return;
+  }
+
+  switch (binding.value->which()) {
+    case mojom::DBValue::Tag::STRING_VALUE: {
+      statement->BindString(binding.index, binding.value->get_string_value());
+      return;
+    }
+    case mojom::DBValue::Tag::INT_VALUE: {
+      statement->BindInt(binding.index, binding.value->get_int_value());
+      return;
+    }
+    case mojom::DBValue::Tag::INT64_VALUE: {
+      statement->BindInt64(binding.index, binding.value->get_int64_value());
+      return;
+    }
+    case mojom::DBValue::Tag::DOUBLE_VALUE: {
+      statement->BindDouble(binding.index, binding.value->get_double_value());
+      return;
+    }
+    case mojom::DBValue::Tag::BOOL_VALUE: {
+      statement->BindBool(binding.index, binding.value->get_bool_value());
+      return;
+    }
+    case mojom::DBValue::Tag::NULL_VALUE: {
+      statement->BindNull(binding.index);
+      return;
+    }
+    default: {
+      NOTREACHED();
+    }
+  }
+}
+
+mojom::DBRecordPtr CreateRecord(
+    sql::Statement* statement,
+    const std::vector<mojom::DBCommand::RecordBindingType>& bindings) {
+  auto record = mojom::DBRecord::New();
+  int column = 0;
+
+  if (!statement) {
+    return record;
+  }
```

Comment:
I know this isn't new code, but is this an expected codepath? Or should we just make this a `DCHECK`?

Reply:
Yes, these should be DCHECKs. I'll update the PR.